server/consumer.go

```diff
@@ -1062,7 +1062,7 @@ func nextReqFromMsg(msg []byte) (time.Time, int, bool, error) {
 		if err := json.Unmarshal(msg, &cr); err != nil {
 			return time.Time{}, -1, false, err
 		}
-		return cr.Expires, cr.Batch, cr.NoWait, nil
+		return time.Now().Add(time.Duration(cr.Expires) * time.Millisecond), cr.Batch, cr.NoWait, nil
```

Comment:
I feel we need to impose a lower limit here maybe? What about a max and default for when none is set?

Reply:
This would be the only place we ever express timeouts in ms..

Reply:
I think we also have nanosecond resolution on duplicate_window, the time stamp of a message, optional start time for playing a stream.

Reply:
start time is a time like the deadline, but I think you may be right on the duplicate window maybe. I am not saying we should not change, I am simply suggesting we do them all at one instead of just this one.

Reply:
I agree. I just shot the ones that I remembered.